### PR TITLE
TIG-3501: Loader and MonotonicLoader actors fails for indexes without name option

### PR DESCRIPTION
**Pull request by mridul-augustine**

### CODE REVIEW

File: src/cast_core/src/Loader.cpp [hunk 1]
```diff
@@ -142,25 +144,59 @@ void genny::actor::Loader::run() {
...
-                        << "Building index " << bsoncxx::to_json(indexKey.view());
                     if (options) {
                         auto indexOptions = (*options)();
-                        BOOST_LOG_TRIVIAL(debug)
-                            << "With options " << bsoncxx::to_json(indexOptions.view());
-                        auto indexOpCtx = _indexBuild.start();
-                        collection.create_index(std::move(indexKey), std::move(indexOptions));
-                        indexOpCtx.success();
+                        bool nameOptionFound = false;
+                        for (auto field : indexOptions.view()) {
```
[the remark this answers is not in the document]
Comment (mridul-augustine) [reply src/cast_core/src/Loader.cpp]: Thanks for this suggestion. This is a much better approach.

**Review by aggrand — APPROVED** [review]
LGTM % a green sys-perf patch for workloads using this!

**Comment by mridul-augustine** [comment]
> LGTM % a green sys-perf patch for workloads using this!

https://spruce.mongodb.com/version/6176fd9d7742ae6ff60bb8b9/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by mridul-augustine** [comment]
evergreen merge

**Comment by mridul-augustine** [comment]
evergreen merge

**Pull request merged**